### DISCUSSION TITLE
Cargo package categories suggestions

### DISCUSSION THREAD

**Discussion by mcskware** [ratatui/ratatui#1034] [Discussion]
tl;dr: Add `categories = ["command-line-interface", "gui"]` to the cargo package metadata in [`Cargo.toml`](https://github.com/ratatui-org/ratatui/blob/main/Cargo.toml).

For people running with the clippy cargo lint [`clippy::cargo-common-metadata`](https://rust-lang.github.io/rust-clippy/master/index.html#cargo_common_metadata) a warning is generated for the main ratatui crate. This is because the cargo package metadata does not currently specify a [`categories`](https://doc.rust-lang.org/cargo/reference/manifest.html#the-categories-field) field.

Note that enabling the clippy group [`clippy::cargo`](https://doc.rust-lang.org/stable/clippy/lints.html#cargo) includes this specific lint by default.

After looking at the [category slugs](https://crates.io/category_slugs) on crates.io, my best guess for the appropriate categories are the [`command-line-interface`](https://crates.io/search?q=category%3Acommand-line-interface) and [`gui`](https://crates.io/search?q=category%3Agui) categories.

As an additional benefit, adding categories to the crate will help discovery on crates.io when users search for these specific slugs.

What I'm hoping to figure out in this discussion is whether categories should be added to the package metadata, and if so, whether the proposed categories above are reasonable. All comments welcome. Cheers!

**Comment by EdJoPaTo** [comment]
I'm not sure about the GUI category as ratatui is terminal based and not graphical so not really a GUI…

Otherwise, yes. Categories should be added.

To the lint: Good idea to run `cargo clippy -- -W clippy::cargo` every once in a while. Sadly it also includes a duplicate dependency versions lint (for good reason) which would break CI runs with it randomly as dependencies update stuff at different times. It would have noticed #961 and wouldn't have stopped complaining all the time until resolved. Personally I don't use it on CI for my personal projects anymore and run it with a local command every once in a while (which still lights up like a Christmas tree as it defines a lot more lints ratatui isn't using (yet?) so I haven't used it on ratatui).

**Comment by mcskware** [comment]
Yup, understood and agreed on both points - a tui is not a gui; and I definitely agree that many if not most projects probably aren't suitable for turning on the cargo lint group by default for CI.

I've created a [pull request](https://github.com/ratatui-org/ratatui/pull/1035) which adds this one-liner.
